src/components/Purchase/CheckoutForm.js

```diff
@@ -0,0 +1,99 @@
+import React, { useEffect, useState } from 'react';
+import {
+  PaymentElement,
+  LinkAuthenticationElement,
+  useStripe,
+  useElements,
+} from '@stripe/react-stripe-js';
+import style from './CheckoutForm.module.css';
+
+export default function CheckoutForm({ paymentAmount }) {
+  const stripe = useStripe();
+  const elements = useElements();
+
+  const [email, setEmail] = useState('');
+  const [message, setMessage] = useState(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isStripeLoaded, setIsStripeLoaded] = useState(false);
+  useEffect(() => {
+    if (!stripe) {
+      return;
+    }
+    // Stripe.js has loaded
+    setIsStripeLoaded(true);
+    const clientSecret = new URLSearchParams(window.location.search).get(
+      'payment_intent_client_secret'
+    );
+
+    if (!clientSecret) {
+      return;
+    }
+  }, [stripe]);
```

Comment:
That's a great question, thanks for asking.
This is important because a PaymentIntent needs to be created on the server before initiating the payment process with Stripe.
By using useEffect with [subtotal] as its dependency array, you ensure that the PaymentIntent is fetched and the clientSecret is updated whenever the subtotal value changes. This is useful in scenarios where the subtotal is dynamic and can change during the user's interaction with the page, and you want to ensure that the latest clientSecret is used when initiating the payment.

Reply:
Ah I see yeah that's a good idea :+1: 

Here it's using `stripe` as the dependency array though? or are you saying that as we build this out with the real data, the idea would be that we'd also have subtotal in there?